design/src/client/orchestrator.md

```diff
@@ -0,0 +1,205 @@
+            Service<Operation<O, Retry>, Response=SdkSuccess<T>, Error=SdkError<E>> + Clone,
+    {
+        self.call_raw(op).await.map(|res| res.parsed)
+    }
+
+    pub async fn call_raw<O, T, E, Retry>(
+        &self,
+        op: Operation<O, Retry>,
+    ) -> Result<SdkSuccess<T>, SdkError<E>>
+        where
+            O: Send + Sync,
+            E: std::error::Error + Send + Sync + 'static,
+            Retry: Send + Sync,
+            R::Policy: bounds::SmithyRetryPolicy<O, T, E, Retry>,
+            Retry: ClassifyRetry<SdkSuccess<T>, SdkError<E>>,
+        // This bound is not _technically_ inferred by all the previous bounds, but in practice it
+        // is because _we_ know that there is only implementation of Service for Parsed
+        // (ParsedResponseService), and it will apply as long as the bounds on C, M, and R hold,
+        // and will produce (as expected) Response = SdkSuccess<T>, Error = SdkError<E>. But Rust
+        // doesn't know that -- there _could_ theoretically be other implementations of Service for
+        // Parsed that don't return those same types. So, we must give the bound.
+            bounds::Parsed<<M as bounds::SmithyMiddleware<C>>::Service, O, Retry>:
+            Service<Operation<O, Retry>, Response=SdkSuccess<T>, Error=SdkError<E>> + Clone,
+    {
+        // The request/response lifecycle
+    }
+}
+```
+
+*The type signature for the new `orchestrate` method:*
+
+```rust
+pub async fn orchestrate(
+    input: Input,
+    runtime_plugins: &RuntimePlugins,
+    // Currently, SdkError is HTTP-only. We currently use it for backwards-compatibility purposes.
+    // The `HttpResponse` generic will likely be removed in the future.
+) -> Result<Output, SdkError<Error, HttpResponse>> {
+	// The request/response lifecycle
+}
+```
+
+Wait a second, I hear you ask. "I see an `Input` and `Output` there, but you're not declaring any generic type arguments. What gives?"
+
+I'm glad you asked. Generally, when you need traits, but you aren't willing to use generic type arguments, then you must [`Box`][std::boxed]. Polymorphism is achieved through [dynamic dispatch] instead of [static dispatch], and this comes with a small runtime cost.
+
+So, what are `Input` and `Output`? They're our own special flavor of a boxed trait object.
+
+```rust
+pub type Input = TypeErasedBox;
+pub type Output = TypeErasedBox;
+pub type Error = TypeErasedBox;
+
+/// A new-type around `Box<dyn Any + Send + Sync>`
+#[derive(Debug)]
+pub struct TypeErasedBox {
+    inner: Box<dyn Any + Send + Sync>,
+}
+```
+
+The orchestrator itself doesn't know about any concrete types. Instead, it passes boxed data between the various components of the request/response lifecycle. Individual components access data in two ways:
+
+- **From the `ConfigBag`:** `let retry_strategy: Box<dyn RetryStrategy> = cfg.get()?;`
+- **From the `InterceptorContext`:** `let put_object_input: PutObjectInput = ctx.input().downcast()?;`
```

Comment:
It would really be:
```rust
let put_object_input: PutObjectInput = ctx.take_input().unwrap().downcast().unwrap()?;
```
Of if you just want a reference:
```rust
let put_object_input = ctx.input().unwrap().downcast_ref::<PutObjectInput>().unwrap();
```